hdinsight-linux-ssh-publickey-metastore-vnet/azuredeploy.json

```diff
@@ -0,0 +1,218 @@
+{
+	"$schema": "https://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#",
+	"contentVersion": "1.0.0.0",
+	"parameters": {
+		"location": {
+			"type": "string",
+			"allowedValues": ["Central US",
+                "East Asia",
+                "East US",
+                "East US 2",
+                "Japan East",
+                "Japan West",
+                "North Europe",
+                "South Central US",
+                "Southeast Asia",
+                "West Europe",
+                "West US"],
+			"metadata": {
+				"description": "The location where all resources will be deployed."
+			}
+		},
+		"clusterName": {
+			"type": "string",
+			"metadata": {
+				"description": "The name of the cluster to create."
+			}
+		},
+        "clusterVersion": {
+            "type": "string",
+            "defaultValue": "3.2",
+			"metadata": {
+				"description": "The HDInsight version to deploy."
+			}
+        },
+        "headNodeSize": {
+            "type": "string",
+            "defaultValue": "Standard_D3",
+            "metadata": {
+				"description": "The VM size of the head nodes."
+			}
+        },
+        "workerNodeSize": {
+            "type": "string",
+            "defaultValue": "Standard_D3",
+            "metadata": {
+				"description": "The VM size of the worker nodes."
+			}
+        },
+		"workerNodeCount": {
+			"type": "int",
+			"defaultValue": 4,
+			"metadata": {
+				"description": "The number of worker nodes in the cluster."
+			}
+		},
+		"clusterLoginUserName": {
+			"type": "string",
+            "defaultValue": "admin",
+			"metadata": {
+				"description": "These credentials can be used to submit jobs to the cluster and to log into cluster dashboards."
+			}
+		},
+		"clusterLoginPassword": {
+			"type": "securestring",
+			"metadata": {
+				"description": "The password must be at least 10 characters in length and must contain at least one digit, one non-alphanumeric character, and one upper or lower case letter."
+			}
+		},
+		"sshUserName": {
+			"type": "string",
+			"metadata": {
+				"description": "These credentials can be used to remotely access the cluster."
+			}
+		},
+		"sshPassword": {
+			"type": "securestring",
+			"metadata": {
+				"description": "The password must be at least 10 characters in length and must contain at least one digit, one non-alphanumeric character, and one upper or lower case letter."
+			}
+		},
+		"clusterStorageAccountName": {
+			"type": "string",
+			"metadata": {
+				"description": "The name of the storage account to use as the cluster's default storage."
+			}
+		},
+        "clusterStorageContainerName": {
+			"type": "string",
+			"metadata": {
+				"description": "The name of the storage container to use."
+			}
+		},
+        "hiveMetastoreServerName": {
```

Comment:
If this is existing, can you add a "existing" prefix to set the right guidance for these parameters